bigtop-deploy/puppet/modules/flink/manifests/init.pp

```diff
@@ -0,0 +1,64 @@
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+class flink {
+
+  class deploy ($roles) {
+    if ("flink-jobmanager" in $roles) {
+      include flink::jobmanager
+    }
+
+    if ("flink-taskmanager" in $roles) {
+      include flink::taskmanager
+    }
+  }
+
+  class common($jobmanager_host, $jobmanager_port, $ui_port) {
+    # make sure flink is installed
+    package { "flink":
+      ensure => latest
+    }
+
+
+    # set values in flink-conf.yaml
+    $tmp_dirs = hadoop::hadoop_storage_dirs
```

Comment:
I'd be much more comfortable if this was passed in as a parameter from the outside of the puppet Flink class. Can you please do it that way?